File: db/src/main/java/org/delivery/db/userorder/UserOrderRepository.java
package org.delivery.db.userorder;

import org.delivery.db.userorder.enums.UserOrderStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface UserOrderRepository extends JpaRepository<UserOrderEntity, Long> {

    // 즉정 유저의 모든 주문
    List<UserOrderEntity> findAllByUserIdAndStatusOrderByIdDesc(Long userId, UserOrderStatus status);

    List<UserOrderEntity> findAllByUserIdAndStatusInOrderByIdDesc(Long userId, List<UserOrderStatus> status);

    // 특정 주문
    Optional<UserOrderEntity> findFirstByIdAndStatusAndUserId(Long id, UserOrderStatus status, Long userId);
}

File: db/src/main/kotlin/org/delivery/db/userorder/UserOrderRepository.kt
package org.delivery.db.userorder

import org.delivery.db.userorder.enums.UserOrderStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserOrderRepository : JpaRepository<UserOrderEntity?, Long?> {
    // 즉정 유저의 모든 주문
    fun findAllByUserIdAndStatusOrderByIdDesc(userId: Long?, status: UserOrderStatus?): MutableList<UserOrderEntity>

    fun findAllByUserIdAndStatusInOrderByIdDesc(
        userId: Long?,
        status: MutableList<UserOrderStatus>?
    ): MutableList<UserOrderEntity>

    // 특정 주문
    fun findFirstByIdAndStatusAndUserId(id: Long?, status: UserOrderStatus?, userId: Long?): UserOrderEntity?

    fun findFirstByIdAndUserId(id: Long?, userId: Long?): UserOrderEntity?
}
